app/celer-sim/CMakeLists.txt

```diff
@@ -40,88 +40,104 @@ if(NOT BUILD_TESTING)
+  endif()
+  
+  add_test(NAME "app/celer-sim-${test_ext}:cpu"
+    COMMAND "${CELER_PYTHON}"
+    "${_driver}" "${_gdml_inp}" "${_hepmc3_inp}" "${mctruth_name}"
+  )
+  set(_env
+    "CELERITAS_DEMO_EXE=$<TARGET_FILE:celer-sim>"
+    "${_geant_exporter_env}"
+    "CELER_LOG=debug"
+    "CELER_LOG_LOCAL=debug"
+    "CELER_DISABLE_DEVICE=1"
+    "CELER_DISABLE_PARALLEL=1"
+    "CELER_CORE_GEO=${CELERITAS_CORE_GEO}"
+  )
+  if(CELERITAS_USE_OpenMP AND CELERITAS_USE_ROOT)
+    # TODO: ROOT writer is not currently thread safe
+    set(CELER_OMP_ENV "OMP_NUM_THREADS=1")
+    set(CELER_PROCESSORS)
+  endif()
+  set_tests_properties("app/celer-sim-${test_ext}:cpu" PROPERTIES
+    ENVIRONMENT "${_env};${CELER_G4ENV};${CELER_OMP_ENV}"
+    REQUIRED_FILES "${_driver};${_gdml_inp};${_hepmc3_inp}"
+    LABELS "app;nomemcheck"
+    ${CELER_PROCESSORS}
+    ${_needs_deps}
+  )
+endfunction()
 
-if(CELERITAS_REAL_TYPE STREQUAL "float")
-  # FIXME: propagation fails in single precision
-  set(_needs_deps DISABLED true)
-endif()
+#-----------------------------------------------------------------------------#
 
-add_test(NAME "app/celer-sim:gpu"
-  COMMAND "${CELER_PYTHON}"
-  "${_driver}" "${_gdml_inp}" "${_hepmc3_inp}" "${_mctruth_out}"
+celer_sim_test(cms
+  simple-cms.gdml gamma-3evt-15prim.hepmc3 simple-cms-mctruth.root
 )
-set_tests_properties("app/celer-sim:gpu" PROPERTIES
-  ENVIRONMENT "${_env};${CELER_G4ENV}"
-  RESOURCE_LOCK gpu
-  REQUIRED_FILES "${_driver};${_gdml_inp};${_hepmc3_inp}"
-  LABELS "app;nomemcheck;gpu"
-  ${_needs_deps}
-  ${_needs_device}
-  ${CELER_NEEDS_PYTHON}
-)
-
-# Enable ROOT output for the celer-sim-cpu test when ROOT is available
-if(CELERITAS_USE_ROOT)
-  set(_mctruth_out "${CMAKE_CURRENT_BINARY_DIR}/mctruth_output.root")
-endif()
-set(CELER_GEO ${CELERITAS_CORE_GEO})
-if(CELERITAS_CORE_GEO STREQUAL "ORANGE" AND NOT CELERITAS_USE_Geant4)
-  # Signal to the app that we have to use the JSON input
-  set(CELER_GEO "${CELER_CORE_GEO}-JSON")
-endif()
 
-add_test(NAME "app/celer-sim:cpu"
-  COMMAND "${CELER_PYTHON}"
-  "${_driver}" "${_gdml_inp}" "${_hepmc3_inp}" "${_mctruth_out}"
-)
-set(_env
-  "CELERITAS_DEMO_EXE=$<TARGET_FILE:celer-sim>"
-  "${_geant_exporter_env}"
-  "CELER_LOG=debug"
-  "CELER_LOG_LOCAL=debug"
-  "CELER_DISABLE_DEVICE=1"
-  "CELER_DISABLE_PARALLEL=1"
-  "CELER_CORE_GEO=${CELERITAS_CORE_GEO}"
-)
-if(CELERITAS_USE_OpenMP AND CELERITAS_USE_ROOT)
-  # TODO: ROOT writer is not currently thread safe
-  set(CELER_OMP_ENV "OMP_NUM_THREADS=1")
-  set(CELER_PROCESSORS)
-endif()
-set_tests_properties("app/celer-sim:cpu" PROPERTIES
-  ENVIRONMENT "${_env};${CELER_G4ENV};${CELER_OMP_ENV}"
-  REQUIRED_FILES "${_driver};${_gdml_inp};${_hepmc3_inp}"
-  LABELS "app;nomemcheck"
-  ${CELER_PROCESSORS}
-  ${_needs_deps}
-)
+# TODO: enable celer-sim-lar[cpu:gpu] test when optical stepping loop is
+#       available. Decide if lar-sphere.gdml should be copied from
+#       test/geocel/data to app/data (as simple-cms.gdml)
```

Comment:
It looks like the two simple-cms geometries are slightly different (the one in app/data has tiny gaps between the cylinders, while the one in test/geocel/data does not). Maybe that's why we have two copies?

Reply:
That is a good point. Could be worth keeping one of them then. The differences most likely come from when the simple-cms geometry was written. I had placed a tolerance between cylinders so we could test boundary crossings between G4 and VecGeom, so we had it updated during geometry tests. Nowadays these numbers are set in our gdml-generator app (see [here](https://github.com/celeritas-project/utils/blob/820af6db40b658793088ab55bba87be91aed899c/gdml-generator/src/SimpleCmsDetector.hh#L51-L57)).

Reply:
I don't think we need the one with the "tolerance": geometry errors etc. should primarily be unit tested rather than the subject of integration tests.